socotratools/socotratools/client.py

```diff
@@ -460,48 +462,57 @@ def get_policy_status(self, policy_locator):
             return 'created'
 
     def create_endorsement(self, policy_locator, endorsement_name,
-                           effective_timestamp, end_timestamp=None,
-                           field_values={}, add_field_groups=[],
-                           update_field_groups=[], remove_field_groups=[],
-                           add_exposures=[], update_exposures=[],
-                           end_exposures=[], action=None):
-
+                           effective_timestamp=None,
+                           field_values=None,
+                           add_field_groups=None,
+                           remove_field_groups=None,
+                           update_field_groups=None,
+                           add_exposures=None,
+                           end_exposures=None,
+                           update_exposures=None,
+                           end_timestamp=None,
```

Comment:
All the `**kwargs` fields above (and including) this one are optional. Two questions.

1. Note: Naming this one `end_timestamp` might be confusing, because it's actually making the endorsement move the effective contract end timestamp of the **policy** (by means of the endorsement) rather than saying "I want to make my endorsement apply to a bounded period of time"--that's possible today by making two endorsements: one with a start/effective timestamp with the "forward" delta, and one with a the timestamp set to the "end" with the "backward" delta that returns the policy to it's original state.

2. Observation: policy "extension" by means of endorsement/renewal always means into the future. AFAIK, it's not desirable (from an industry perspective--please correct me if I'm wrong) for us to support endorsements that move a **policy's contract start timestamp**, effectively mutating the risk on-book retroactively into the past. Would one rather cancel errors (or endorse to add/remove coverage,) from some existing policy?

3. I noticed the `reprice` option isn't supported here. Probably intentional.